File: src/test/java/com/zeek/javatest/arraylist/Person.java
package com.zeek.javatest.arraylist;

/**
 * @author: weibo_li
 * @since: 2018-03-06 下午7:56
 */
public class Person {

    static {
        System.out.println("静态代码块");
    }

    Person() {
        System.out.println("构造函数");
        System.out.println(age);
    }

    {
        System.out.println("非静态代码块");
    }



    Integer age;
}

File: src/test/java/com/zeek/kotlintest/classtest/Person.kt
package com.zeek.kotlintest.classtest

/**
 * 定义属性和方法
 * @author: weibo_li
 * @since: 2018-11-26 下午3:31
 */
class Person {

    //定义两个属性
    //var 属性自动生成getter和setter
    var name : String = ""
    //val 属性只生成getter
    val age : Int = 0

    //定义一个方法
    fun say(content : String) {
        println(content)
    }

}

fun main(args: Array<String>) {
    var p : Person = Person()

    //在Java代码中可以访问kotlin自动为Person生成的getter和setter方法，但是在kotlin程序中不能调用getter和setter方法

}

